components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsAdapter.kt

```diff
@@ -0,0 +1,218 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons.ui
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.RatingBar
+import android.widget.TextView
+import androidx.annotation.StringRes
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import mozilla.components.feature.addons.Addon
+import mozilla.components.feature.addons.R
+import mozilla.components.feature.addons.amo.AddonCollectionProvider
+import mozilla.components.feature.addons.ui.CustomViewHolder.AddonViewHolder
+import mozilla.components.feature.addons.ui.CustomViewHolder.SectionViewHolder
+import mozilla.components.feature.addons.ui.CustomViewHolder.UnsupportedSectionViewHolder
+
+private const val VIEW_HOLDER_TYPE_SECTION = 0
+private const val VIEW_HOLDER_TYPE_NOT_YET_SUPPORTED_SECTION = 1
+private const val VIEW_HOLDER_TYPE_ADDON = 2
+
+/**
+ * An adapter for displaying add-on items.
+ */
```

Comment:
nit: Let's elaborate a bit here. This adapter will show information related to the state of an add-on, like if it is recommend, unsupported or already installed, additionally will allow to perform actions like installing.

Reply:
Fixed